pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -298,6 +299,38 @@ def test_iloc_setitem_bool_indexer(self, klass):
         expected = DataFrame({"flag": ["x", "y", "z"], "value": [2, 3, 4]})
         tm.assert_frame_equal(df, expected)
 
+    @pytest.mark.parametrize("dtype", ["int64", "Int64"])
+    def test_setitem_complete_columns_different_dtypes(self, dtype):
+        # GH: 20635
+        df = DataFrame({"A": ["a", "b"], "B": ["1", "2"], "C": ["3", "4"], "D": [1, 2]})
+        df.loc[:, ["B", "C"]] = df.loc[:, ["B", "C"]].astype("int64").astype(dtype)
+        expected = DataFrame({"A": ["a", "b"], "B": [1, 2], "C": [3, 4], "D": [1, 2]})
+        expected[["B", "C"]] = expected[["B", "C"]].astype(dtype)
+        tm.assert_frame_equal(df, expected)
+
+    def test_setitem_single_column_as_series_different_dtype(self):
+        # GH: 20635
+        df = DataFrame({"A": ["a", "b"], "B": ["1", "2"], "C": ["3", "4"]})
+        df.loc[:, "C"] = df.loc[:, "C"].astype("int64")
+        expected = DataFrame({"A": ["a", "b"], "B": ["1", "2"], "C": [3, 4]})
+        tm.assert_frame_equal(df, expected)
+
+    def test_setitem_conversion_to_datetime(self):
+        # GH: 20511
+        df = DataFrame(
+            [["2015-01-01", "2016-01-01"], ["2016-01-01", "2015-01-01"]]
+        ).add_prefix("date")
```

Comment:
can you do this without add_prefix?  try to keep it as obvious as possible what is being tested

Reply:
Done